earthaccess/virtualizarr.py

```diff
@@ -0,0 +1,112 @@
+from __future__ import annotations
+
+import fsspec
+import xarray as xr
+
+import earthaccess
+
+
+def _parse_dmr(
+    fs: fsspec.AbstractFileSystem,
+    data_path: str,
+    dmr_path: str = None
+) -> xr.Dataset:
+    """
+    Parse a granule's DMR++ file and return a virtual xarray dataset
+
+    Parameters
+    ----------
+    granule : earthaccess.results.DataGranule
+        The granule to parse
+        fs : fsspec.AbstractFileSystem
+        The file system to use to open the DMR++
+
+    Returns
+    ----------
+    xr.Dataset
+    The virtual dataset (with virtualizarr ManifestArrays)
+
+    Raises
+    ----------
+    Exception
+    If the DMR++ file is not found or if there is an error parsing the DMR++
+    """
+    from virtualizarr.readers.dmrpp import DMRParser
```

Comment:
It's interesting that you're not actually using the `filetype='dmr++'` option to `virtualizarr.open_virtual_dataset` here. It seems to me that one alternative option would be for everything in https://github.com/zarr-developers/VirtualiZarr/pull/113 to also live in this library, as it already pretty much entirely uses public virtualizarr API... But I guess that depends whether you think the dmr++ option to `virtualizarr.open_virtual_dataset` is likely to be useful outside of the context of the `earthaccess` library.

Reply:
The only reason is because the parser has an additional kwarg `data_filepath` which is required in cases where the dmr path cannot be simply derived by just adding `.dmrpp`. If there is a way for `engine` specific args in `virtualizarr.open_dataset` then I can switch to that

Reply:
> The only reason is because the parser has an additional kwarg data_filepath which is required in cases where the dmr path cannot be simply derived by just adding .dmrpp.

I'm not sure I understand - why is the main `filepath` you pass to `virtualizarr.open_virtual_dataset` not sufficient?

Reply:
For this case: `virtualizarrr.open_dataset(filepath=“s3://air.dmrpp”, data_filepath=“s3://datafiles/air.nc”, engine="dmr++")` when the dmr path is independent from data path. The chunk manifest needs to store the `data_filepath` instead of the dmr filepath

Reply:
Huh - does the dmr++ data not contain the path to the original data??

Reply:
No the dmr file only contains the file name and not the full path. This is an example of the name that a dmr file contains `name="20210715090000-JPL-L4_GHRSST-SSTfnd-MUR-GLOB-v02.0-fv04.1.nc"`

Reply:
I noticed that [virtualizarr renaming paths](https://virtualizarr.readthedocs.io/en/latest/generated/virtualizarr.xarray.VirtualiZarrDatasetAccessor.rename_paths.html#virtualizarr.xarray.VirtualiZarrDatasetAccessor.rename_paths) was added and I think this can solve my issue. I will just call `vz.open_dataset` and then rename the data paths using `earthaccess` results. Then I can switch to the public `virtualizarr` API now and remove the `_parse_dmr` function

Reply:
So what does that imply for my original question:

> It seems to me that one alternative option would be for everything in https://github.com/zarr-developers/VirtualiZarr/pull/113 to also live in this library, as it already pretty much entirely uses public virtualizarr API...

Reply:
@betolink How about if I move the parser code into `earthaccess`? Now that I'm thinking about it makes more sense to be in the NASA related repository. It would also make unit tests easier since `earthaccess` can easily access NASA dmrpp files. Then this PR will add `dmrpp.py` and `virtualizarr.py`